doc/website/source/guides/widgets.md

```diff
@@ -0,0 +1,132 @@
+
+If `useSteps` is true for the slide configuration, the bullet points will be rendered one by one as the user steps through the slide.
+
+You can use `stepOffset` to offset the step number for the first item in the list. For instance, if you set `stepOffset` to 1, the first item will only be visible when the user steps to the second step. Remember to increase the number of `steps` for the slide configuration if you use `stepOffset`.
+
+```dart
+class FlutterDeckBulletListDemoSlide extends FlutterDeckSlideWidget {
+  const FlutterDeckBulletListDemoSlide()
+      : super(
+          configuration: const FlutterDeckSlideConfiguration(
+            route: '/bullet-list-demo',
+            title: 'Bullet list demo',
+            steps: 3, // Define the number of steps for the slide
+          ),
+        );
+
+  @override
+  FlutterDeckSlide build(BuildContext context) {
+    return FlutterDeckSlide.split(
+      leftBuilder: (context) => FlutterDeckBulletList(
+        useSteps: true, // Enable steps for the bullet list
+        items: const [
+          'This is a step',
+          'This is another step',
+          'This is a third step',
+        ],
+      ),
+      rightBuilder: (context) => const Text('FlutterDeckBulletList demo'),
+    );
+  }
+}
+```
+
+### FlutterDeckCodeHighlight
+
+Provides a widget that gives you customizable syntax highlighting for many languages.
+
+```dart
+class CodeHighlightSlide extends FlutterDeckSlideWidget {
+  const CodeHighlightSlide()
+      : super(
+          configuration: const FlutterDeckSlideConfiguration(
+            route: '/code-highlight',
+            header: FlutterDeckHeaderConfiguration(title: 'Code Highlighting'),
+          ),
+        );
+
+  @override
+  FlutterDeckSlide build(BuildContext context) {
+    return FlutterDeckSlide.blank(
+      builder: (context) => const Center(
+        child: FlutterDeckCodeHighlight(
+          code: '''
+import 'package:flutter/material.dart';
+import 'package:flutter_deck/flutter_deck.dart';
+
+class CodeHighlightSlide extends FlutterDeckSlideWidget {
+  const CodeHighlightSlide()
+      : super(
+          configuration: const FlutterDeckSlideConfiguration(
+            route: '/code-highlight',
+            header: FlutterDeckHeaderConfiguration(title: 'Code Highlighting'),
+          ),
+        );
+
+  @override
+  FlutterDeckSlide build(BuildContext context) {
+    return FlutterDeckSlide.blank(
+      builder: (context) => const Center(
+        child: Text('Use FlutterDeckCodeHighlight widget to highlight code!'),
+      ),
+    );
+  }
+}''',
+          fileName: 'code_highlight_slide.dart',
+          language: 'dart',
+        ),
+      ),
+    );
+  }
+}
+```
+
+![FlutterDeckCodeHighlight](https://github.com/mkobuolys/flutter_deck/blob/main/images/code_highlighting.png?raw=true)
+
+## 📦 Accessing slide deck state from the code
```

Comment:
This section should probably have a dedicated page under Guides.

Reply:
Should it? Seems like a small amount of info that's highly related to widgets. I'll move it if you'd like, but wanted to double check.

It's also possible that "widgets" isn't the best organizing concept in the first place. We could also take pieces from this file and move them to other guides. Let me know what's best.